api/scpca_portal/models/job.py

```diff
@@ -224,46 +212,76 @@ def terminate_submitted(cls) -> List[Self]:
             for job in jobs:
                 if batch.terminate_job(job):
                     job.state = JobStates.TERMINATED.value
-                    job.terminated_at = make_aware(datetime.now())
+                    job.completed_at = make_aware(datetime.now())
                     terminated_jobs.append(job)
 
-            Job.objects.bulk_update(terminated_jobs, ["state", "terminated_at"])
+            Job.objects.bulk_update(terminated_jobs, ["state", "completed_at"])
 
         return terminated_jobs
 
-    # NOTE: This will be refactored later (e.g., save itself before job submission for job.id)
+    @classmethod
+    def bulk_update_state(cls, synced_jobs: List[Self]):
+        """
+        Updates the states of the synced jobs and their associated datasets.
+        """
+        cls.objects.bulk_update(
+            synced_jobs,
+            ["state", "failure_reason", "completed_at"],
+        )
+
+        Dataset.apply_last_jobs(synced_jobs)
```

Comment:
Pass datasets here.

Reply:
I've left it as is and created a mapping object to retrieve the corresponding jobs from `update_from_last_jobs`. Let me know your thoughts on this!